software/test_project/src/map_builder/Map.java

```diff
@@ -119,4 +119,30 @@ private void updateEntitiesInMap(ArrayList<Entity> entities){
 	}
 	
 
+
+
+
+
+
+	// BEWARE=========================================================================================
+//====================================================================================================
+// 						DANGER
+	// KEEP OUT
+	// SASHA`S CODE
+
+	private void DrawMapFromFile(String MapFileName){
+		char[][] mapArr= MapSaver.readMap(MapFileName);
+		for(int col=0; col<Constants.WINDOW_MAP_WIDTH; col+=4){
```

Comment:
'col' should not be incremented by 4.
Rather it should be incremented by 1... according to the grid!